### DISCUSSION TITLE
Provide the ability to choose the resolution of Asset Store images

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#2577] [Enhancement ideas (and their technical discussions)]
### Problem ###

We currently import image assets into the Asset Store based on guidelines that try to balance quality with efficiency.  High-resolution images take more resources, but they enable use-cases where large images are needed.  Smaller images enable a game to run with fewer resources.  

The current Asset Store import process stores a single image with a medium-quality resolution.  If this image size is too small for the user, they must scale it up inside the game engine (with loss in quality) or they can download the asset outside the Asset Store.   If the Asset Store image size is too large, the user must scale it down.

### Proposed Solution ###

1. The Asset Store process should store a copy of each image at its full resolution (along with a thumbnail for the Asset Store interface).

2. When a user selects an image from the Asset Store, they are asked what image resolution is desired.  This process might list common sizes (small, medium, large, etc) but also lets users enter custom pixel dimensions.  GDevelop will not let the user pick a resolution higher than the resolution of the original file. 

3. Once the user selects the size they want, the full-size image is downloaded and resized by the GDevelop client.  The full-size image is locally cached in case the user wants to resize the image in the future.

### Considerations  ###

This method provides the most flexibility for users, but it has some downsides:

- Users will slow Internet speeds will have to wait longer to download the full resolution files
- GDevelop will incur more cloud costs for storage and transfer
-- GDevelop could require a subscription in order to use the Asset Store to pay for the costs of the service

**Comment by 4ian** [comment]
Agree that we could scale down images locally if needed, though this also implies updating objects points and collision masks.
This could actually be a generic feature for objects in the editor, to scale down the resources used and adapt the object properties.
